### PR TITLE
feat(telemetry): add log sanitization helper for user-controlled data

### LINKED ISSUE #3729: feat(telemetry): add log sanitization helper for user-controlled data
## Background

PR #3728 added Context Manager telemetry for file selection observability. Gemini Code Assist identified potential log injection vulnerabilities where user-controlled data is logged without sanitization.

## Current State

The following telemetry functions log user-controlled data:
- `log_context_file_scan()`: logs `goal` parameter (user's goal text)
- `log_context_file_select()`: logs `selected_files` (file paths)
- `log_context_token_budget()`: logs `excluded_files` (file paths)

## Risk Assessment

**LOW** - The risk is mitigated by:
1. `goal` comes from internal LLM planner, not directly from user input
2. File paths come from repository filesystem (controlled environment)
3. Logs use structured JSON format (`extra` dict), not plain text
4. Modern log aggregators handle structured logs safely

## Recommendation

Add a sanitization helper function to `resource_telemetry.py`:

```python
def _sanitize_for_log(value: str) -> str:
    """Sanitize string for safe logging (remove newlines, control chars)"""
    if not value:
        return ""
    return value.replace('\n', ' ').replace('\r', ' ')
```

Apply to:
- `goal_preview` in `log_context_file_scan()`
- File paths in `log_context_file_select()` and `log_context_token_budget()`

## Blueprint Alignment

- Section 4.2 Telemetry v2: Telemetry should be secure and not introduce vulnerabilities
- Defense in depth: Even if risk is low, sanitization adds an extra layer of protection

## References

- PR #3728: https://github.com/RC918/morningai/pull/3728
- Gemini Code Assist comments: https://github.com/RC918/morningai/pull/3728#discussion_r2675001292

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Implements Issue #3729 - adds a `_sanitize_for_log()` helper function to `resource_telemetry.py` that removes newlines and carriage returns from user-controlled data before logging. This provides defense-in-depth against log injection attacks.

**Changes:**
- Added `_sanitize_for_log(value: str) -> str` helper that replaces `\n` and `\r` with spaces
- Applied sanitization to `goal_preview` in `log_context_file_scan()`
- Applied sanitization to file paths in `log_context_file_select()`
- Applied sanitization to `excluded_files` in `log_context_token_budget()`
- Added 18 unit tests covering edge cases and integration with telemetry functions

Fixes #3729

## Updates since last revision

- Fixed `test_catches_mixed_indentation` test to handle Python version differences (some versions raise `TabError`, others raise `SyntaxError` with "tabs and spaces" message)

## Review & Testing Checklist for Human

- [ ] Verify the double-space behavior for CRLF (`\r\n` → `"  "`) is acceptable for your log aggregation system
- [ ] Confirm no other user-controlled data in telemetry functions needs sanitization
- [ ] Review whether additional control characters (tabs, backspaces) should be sanitized in a follow-up

**Test plan:** Run `pytest handoff/20250928/40_App/orchestrator/tests/test_resource_telemetry.py -v` to verify all 18 tests pass.

### Notes

Requested by: @RC918
Link to Devin run: https://app.devin.ai/sessions/a786a19a99684dcea55b22bf7b175d94

### CODE REVIEW

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review

File: handoff/20250928/40_App/orchestrator/resource_telemetry.py [hunk 1]
```diff
@@ -41,6 +41,27 @@ def _is_telemetry_enabled() -> bool:
...
+    characters that could be used to forge log entries or inject malicious
+    content into log aggregators.
+
+    Issue #3729: Log sanitization helper for user-controlled data.
+
+    Args:
+        value: String to sanitize (may contain user-controlled data)
+
+    Returns:
+        Sanitized string with newlines and carriage returns replaced by spaces
```
[the remark this answers is not in the document]
Comment (RC918) [reply handoff/20250928/40_App/orchestrator/resource_telemetry.py]: The docstring already mentions the security purpose in lines 45-48:

> "This provides defense-in-depth against log injection attacks by removing characters that could be used to forge log entries or inject malicious content into log aggregators."

This explicitly states the function's security purpose for log injection protection.

File: handoff/20250928/40_App/orchestrator/resource_telemetry.py [hunk 2]
```diff
@@ -41,6 +41,27 @@ def _is_telemetry_enabled() -> bool:
...
+    """
+    if not value:
+        return ""
+    return value.replace('\n', ' ').replace('\r', ' ')
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/resource_telemetry.py]: IMPACT: The current implementation only removes newlines and carriage returns. Suggested FIX: Consider expanding the sanitization to remove other control characters that could be used in log injection attacks (e.g., ASCII control codes 0-31).

File: handoff/20250928/40_App/orchestrator/tests/test_resource_telemetry.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        """Carriage returns should be replaced with spaces."""
+        assert _sanitize_for_log("line1\rline2") == "line1 line2"
+
+    def test_removes_crlf(self):
+        """CRLF sequences should be replaced with spaces."""
+        assert _sanitize_for_log("line1\r\nline2") == "line1  line2"
+
+    def test_log_injection_attempt(self):
+        """Should prevent log injection attacks."""
+        malicious = "normal text\n[FAKE_EVENT] Injected log entry"
```
[the remark this answers is not in the document]
Comment (RC918) [reply handoff/20250928/40_App/orchestrator/tests/test_resource_telemetry.py]: The test focuses on log injection attacks (forging new log entries via newlines), which is the scope of Issue #3729.

Characters like `%`, `{`, `}` are related to format string vulnerabilities, which is a different vulnerability class. Our telemetry uses structured logging with the `extra` dict, so format string attacks aren't applicable here - the user-controlled data goes into JSON fields, not the format string itself.
